fw/http_msg.c

```diff
@@ -815,6 +820,33 @@ __hdr_sub(TfwHttpMsg *hm, const TfwStr *hdr, unsigned int hid)
 	return TFW_PASS;
 }
 
+/**
+ * Substitute request method with "get"
+ *
+ * Note: replaced method name length is supposed to be shorter
+ *       than the replacement string ("get")
+ */
+int
+tfw_http_req_meth_sub_with_get(TfwHttpMsg *hm, unsigned char dest_meth)
+{
+	const char *dest_name;
+	size_t dest_len;
+
+	TfwStr sub_str = TFW_STR_STRING("get");
+
+	BUG_ON(hm->req->method != dest_meth);
+
+	dest_name = tfw_http_tbl_method_get_name(dest_meth);
+	dest_len = strlen(dest_name);
```

Comment:
I think this is the case where it can be done more effectively at compile time. A static table can be created that has the length of the string for each HTTP method, like `[TFW_HTTP_METH_PURGE] = SLEN("PURGE")`. There was no need for that before, but it looks like now there's a good case for that. There's a similar table for headers.

Reply:
ok. I made it plain replace PURGE with GET, no variants.
just at the beginning there was a discussion to create a generic function to rewrite. then we simplified it to multiple targets - single subst. now it's simply designed for one case.

Reply:
done